# DESAFIO 1

In [1]:
print("Hola Mundo, esta es mi primera incursión en Python")

Hola Mundo, esta es mi primera incursión en Python


# DESAFIO 2

## ITEM 1

In [27]:
name = "Gabriel Arístides Camilo San Martín"

## ITEM 2

In [ ]:
edad = 28

## ITEM 3

In [ ]:
actividades_favoritas = ['viajar', 'ver peliculas', 'dormir']

## ITEM 4

In [9]:
mascota = True

# DESAFIO 3

## ITEM 1

In [28]:
print(name)
print(edad)
print(actividades_favoritas[1])
print(mascota)

Gabriel Arístides Camilo San Martín
28
ver peliculas
True


## ITEM 2

In [29]:
print(type(name))
print(type(edad))
print(type(actividades_favoritas))
print(type(mascota))

<class 'str'>
<class 'int'>
<class 'list'>
<class 'bool'>


## ITEM 3

In [30]:
presentacion = 'Mi nombre es ' + name + ' y tengo ' + str(edad) + ' años. Mis actividades favoritas son '
contador = 0
for actividad in actividades_favoritas:
    contador += 1
    presentacion += actividad
    if contador == 1:
        presentacion += ', '
    elif contador == 2:
        presentacion += ' y '
if mascota:
    presentacion += '. Tengo una mascota'
else:
    presentacion += '. No tengo una mascota'
print(presentacion)

Mi nombre es Gabriel Arístides Camilo San Martín y tengo 28 años. Mis actividades favoritas son viajar, ver peliculas y dormir. Tengo una mascota


# DESAFIO 4

## ITEM 1

In [19]:
print('Estaba la pájara pinta sentada en el verde limón')


Estaba la pájara pinta sentada en el verde limón


Faltaba el cierre del string

## ITEM 2

In [31]:
print('Mi nombre es ' + name + ' y tengo ' + str(edad) + ' años')

Mi nombre es Gabriel Arístides Camilo San Martín y tengo 28 años


se debe agregar el signo "+" para concatenar los string. se añade espacios despues de la palabre "es" y "tengo" y antes de "y" y "años". finalmente la variable edad es un int que debe ser parseado a string

## ITEM 3

In [36]:
import pandas as pd
import numpy as np

Las librerias estan mal escritas

## ITEM 4

In [ ]:
"Ornitorrinco" + str(45)

se debe parsear a string el numero 45

# DESAFIO 5

## ITEM 1

In [35]:
import pandas as pd
import numpy as np

## ITEM 2

In [33]:
df = pd.read_csv('flights.csv')

## ITEM 3

In [43]:
print (df[:5])
print (df[-5:])

   year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129
4  1949       May         121
     year      month  passengers
139  1960     August         606
140  1960  September         508
141  1960    October         461
142  1960   November         390
143  1960   December         432


## ITEM 4

In [45]:
print(df['year'].describe())

count     144.000000
mean     1954.500000
std         3.464102
min      1949.000000
25%      1951.750000
50%      1954.500000
75%      1957.250000
max      1960.000000
Name: year, dtype: float64


## ITEM 5

In [49]:
print(df['year'].value_counts())
print(df['month'].value_counts())

1960    12
1959    12
1958    12
1957    12
1956    12
1955    12
1954    12
1953    12
1952    12
1951    12
1950    12
1949    12
Name: year, dtype: int64
March        12
June         12
May          12
November     12
August       12
September    12
December     12
January      12
October      12
April        12
February     12
July         12
Name: month, dtype: int64


## ITEM 6

In [76]:
primeras = df[:15]
ultimas = df[-15:]
print('Primeras 15')
print('-----------')
print(primeras)
print('\nUltimas 15')
print('-----------')
print(ultimas)

Primeras 15
-----------
    year      month  passengers
0   1949    January         112
1   1949   February         118
2   1949      March         132
3   1949      April         129
4   1949        May         121
5   1949       June         135
6   1949       July         148
7   1949     August         148
8   1949  September         136
9   1949    October         119
10  1949   November         104
11  1949   December         118
12  1950    January         115
13  1950   February         126
14  1950      March         141

Ultimas 15
-----------
     year      month  passengers
129  1959    October         407
130  1959   November         362
131  1959   December         405
132  1960    January         417
133  1960   February         391
134  1960      March         419
135  1960      April         461
136  1960        May         472
137  1960       June         535
138  1960       July         622
139  1960     August         606
140  1960  September         508
141  1960  

## ITEM 7

In [81]:
print('########MEDIA########')
print('Muestra Completa')
print('----------------')
print(np.mean(df['passengers']))
print('\nPrimeras 15')
print('-----------')
print(np.mean(primeras['passengers']))
print('\nÚltimas 15')
print('-----------')
print(np.mean(ultimas['passengers']))

print('\n#######MEDIANA#######')
print('Muestra Completa')
print('----------------')
print(np.median(df['passengers']))
print('\nPrimeras 15')
print('-----------')
print(np.median(primeras['passengers']))
print('\nÚltimas 15')
print('-----------')
print(np.median(ultimas['passengers']))

print('\n#DESVIACIÓN ESTANDAR#')
print('Muestra Completa')
print('----------------')
print(np.std(df['passengers'], ddof = 1 ))
print('\nPrimeras 15')
print('-----------')
print(np.std(primeras['passengers'], ddof = 1 ))
print('\nÚltimas 15')
print('-----------')
print(np.std(ultimas['passengers'], ddof = 1 ))


########MEDIA########
Muestra Completa
----------------
280.2986111111111

Primeras 15
-----------
126.8

Últimas 15
-----------
459.2

#######MEDIANA#######
Muestra Completa
----------------
265.5

Primeras 15
-----------
126.0

Últimas 15
-----------
432.0

#DESVIACIÓN ESTANDAR#
Muestra Completa
----------------
119.9663169429432

Primeras 15
-----------
13.126853610606226

Últimas 15
-----------
77.93697087115606


Existe una gran diferencia entre las medidas de tendencia central (media y mediana) y dispersión calculadas a partir de todas las observaciones, las primeras 15 observaciones y las últimas 15 observaciones respectivas. 

En las primeras 15 observaciones media y mediana son relativamente similares (126) y presentan una desviación estándar relativamente baja (13,127 aprox.). Por su parte las ultimas 15 observaciones presentan una diferencia significativa de media y median, y además la desviación estándar es alta. Esto se ve reflejado en las medidas de todas las observaciones, la cual se ve fuertemente influenciada por las últimas observaciones quitándole homogeneidad. 
